### PR TITLE
Unify functions for spike positions and spike positions just before the ROI

### PR DESCRIPTION
**Pull request by maxnoe**
Removes duplicated code by refactoring into a common function

### CODE REVIEW

File: ctapipe_io_lst/calibration.py [hunk 1]
```diff
@@ -556,100 +556,104 @@ def interpolate_spike_A(waveform, position):
...
+    if last_capacitor % 2 != 0 or last_capacitor > last_in_first_half:
+        # bad trickery to get numba to compile an empty list with type int
+        # see https://numba.pydata.org/numba-doc/latest/user/troubleshoot.html
+        return [int(x) for x in range(0)]
+
+    # we have two cases for spikes that can occur in each of the 4 channels
+    base_positions = (
+        N_CAPACITORS_PIXEL - N_SAMPLES - previous_first_cap - shift,
+        previous_first_cap + N_SAMPLES - shift
+    )
```
Comment (SeiyaNozaki) [inline ctapipe_io_lst/calibration.py:591]: I'd suggest using `last_capacitor` defined above to make it easier to understand.
```
best_positions = (
    N_CAPACITORS_PIXEL - last_capacitor - 2 - shift,
    last_capacitor - shift
)
```
and `shift` is 0, 1 for the current and old firmware respectively.
What do you think？

**Review by SeiyaNozaki — COMMENTED** [review]
I think new lines to get spike positions with `shift` parameters are correct, and this PR seems fine.
I added a suggestion to make it easier to understand the spike conditions, but not a strong request.

File: ctapipe_io_lst/calibration.py (hunk 1 above)
Comment (maxnoe) [reply ctapipe_io_lst/calibration.py:591]: Perfect

**Comment by maxnoe** [comment]
@jsitarek The code now produces the spike positions in the range [-2, 39]

**Review by SeiyaNozaki — APPROVED** [review]
It seems good, thanks!

**Pull request merged**